### PR TITLE
Allows for vttestserver and vtcombo to respond to VtGateExecute.

### PR DESCRIPTION
**Pull request by mavenraven**
Currently, if you try to use `vtctlclient VtGateExecute` against `vttestserver`, it doesn't work. This fixes that.

### CODE REVIEW

**Comment by mavenraven** [comment]
So there's some test pollution in `vttestserver_test.go`. I thought I had introduced it, but it's actually something with the mTLS tests. When my test was at the bottom of the file, it would hang forever. Where it's at right now, it's ok. Here's a minimum example of the issue: 
https://github.com/vitessio/vitess/pull/7127

I couldn't figure it out after digging in for a half hour, and I have other stuff that's higher priority to work on, so I figured maybe someone else could figure out what the root cause of it is, if we even care. Anyways, this should be good to go now.

File: go/cmd/vttestserver/vttestserver_test.go [hunk 1]
```diff
@@ -103,7 +147,7 @@ func TestMtlsAuth(t *testing.T) {
 	assertColumnVindex(t, cluster, columnVindex{keyspace: "app_customer", table: "customers", vindex: "hash", vindexType: "hash", column: "id"})
 }
 
-func TestMtlsAuthUnaothorizedFails(t *testing.T) {
+func TestMtlsAuthUnauthorizedFails(t *testing.T) {
```
Comment (deepthi) [inline go/cmd/vttestserver/vttestserver_test.go:150]: +1. That bothered me when I first saw it :)

**Review by deepthi — APPROVED** [review]
LGTM

File: go/cmd/vttestserver/vttestserver_test.go [hunk 2]
```diff
@@ -62,6 +65,47 @@ func TestRunsVschemaMigrations(t *testing.T) {
...
+			"select 'success';",
+		},
+		30*time.Second,
+	)
+	assert.NoError(t, err)
+
+	var b strings.Builder
+	b.Grow(1024)
+
+OUT:
```
Comment (deepthi) [inline go/cmd/vttestserver/vttestserver_test.go:93]: A tiny nit: the linter doesn't care but I believe it is common to name labels like variables. So `out` or `Out`.

**Pull request merged**